clippy_lints/src/doc/mod.rs

```diff
@@ -502,14 +551,18 @@ fn check_attrs(cx: &LateContext<'_>, valid_idents: &FxHashSet<String>, attrs: &[
     suspicious_doc_comments::check(cx, attrs);
 
     let (fragments, _) = attrs_to_doc_fragments(attrs.iter().map(|attr| (attr, None)), true);
-    let mut doc = String::new();
-    for fragment in &fragments {
-        add_doc_fragment(&mut doc, fragment);
-    }
+    let mut doc = fragments.iter().fold(String::new(), |mut acc, fragment| {
+        add_doc_fragment(&mut acc, fragment);
+        acc
+    });
     doc.pop();
+    let doc = doc.trim().to_string();
 
     if doc.is_empty() {
```

Comment:
Could go with `if doc.chars().all(char::is_whitespace) {` rather than trimming `doc`, I don't know if it matters for rustdoc but leading/trailing whitespace is significant in markdown sometimes. Even if not it saves us an allocation